### DISCUSSION TITLE
MySQL Workbench called LearningSQL;

### DISCUSSION THREAD

**Discussion by Faithsakho** [community/community#76143] [Programming Help]
### Body

Answer the following questions using Joins in SQL. Please note that there might be several different ways to join the tables (e.g., inner join, where, nested query, etc.). 

1. Create a list of all employees first and last names, and the name of the branch. Use  INNER JOIN ON syntax. 
2. Using an inner join, show the number of employees in each department, grouped by department. 
3. Using a one-sided outer join, show the number of employees in each department.
4. Show a list of the total balance for each Branch, including the branch name (using inner join).
5. Retrieve a list of employees who worked with individual customers (i.e., CUST_TYPE_CD = ‘I’). Display those employees’ last name, first name, title, and respective CustomerID.  (Hint: joining account, employee, and customer)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by MegatIrfan** [comment]
Answer Question 1 : 
SELECT e.FirstName, e.LastName, b.BranchName
FROM Employees e
INNER JOIN EmployeeDepartment ed ON e.EmployeeID = ed.EmployeeID
INNER JOIN Departments d ON ed.DepartmentID = d.DepartmentID
INNER JOIN Branches b ON d.BranchID = b.BranchID;

Anwer Question 2 : 
SELECT d.DepartmentName, COUNT(e.EmployeeID) AS EmployeeCount
FROM Employees e
INNER JOIN EmployeeDepartment ed ON e.EmployeeID = ed.EmployeeID
INNER JOIN Departments d ON ed.DepartmentID = d.DepartmentID
GROUP BY d.DepartmentName;

Answer Question 3 : 
SELECT d.DepartmentName, COUNT(e.EmployeeID) AS EmployeeCount
FROM Departments d
LEFT JOIN EmployeeDepartment ed ON d.DepartmentID = ed.DepartmentID
LEFT JOIN Employees e ON ed.EmployeeID = e.EmployeeID
GROUP BY d.DepartmentName;

Answer Question 4 : 
SELECT b.BranchName, SUM(a.Balance) AS TotalBalance
FROM Branches b
INNER JOIN Departments d ON b.BranchID = d.BranchID
INNER JOIN EmployeeDepartment ed ON d.DepartmentID = ed.DepartmentID
INNER JOIN Employees e ON ed.EmployeeID = e.EmployeeID
INNER JOIN Accounts a ON e.EmployeeID = a.EmployeeID
GROUP BY b.BranchName;

Answer Question 5 : 
SELECT e.LastName, e.FirstName, e.Title, a.CustomerID
FROM Employees e
INNER JOIN Accounts a ON e.EmployeeID = a.EmployeeID
INNER JOIN Customers c ON a.CustomerID = c.CustomerID
WHERE c.CUST_TYPE_CD = 'I';

**Reply by Faithsakho** [reply]
Thank you but only the first answers worked. any idea on the others
